src/Storage/DataSet.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * This file is part of Lcobucci\JWT, a simple library to handle JWT and JWS
+ *
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD-3-Clause
+ */
+
+declare(strict_types=1);
+
+namespace Lcobucci\JWT\Storage;
+
+use OutOfBoundsException;
+
+/**
+ * @author Luís Otávio Cobucci Oblonczyk <lcobucci@gmail.com>
+ *
+ * @since 4.0.0
+ */
+final class DataSet
+{
+    /**
+     * @var array
+     */
+    private $data;
+
+    /**
+     * @var string
+     */
+    private $payload;
+
+    public function __construct(array $data, string $payload)
+    {
+        $this->data = $data;
```

Comment:
There's no guarantee that `$data` is json-serializable here, right? What does `$data` have to respect?